test/SILOptimizer/address_lowering.sil

```diff
@@ -1170,6 +1171,85 @@ bb2(%9 : $Error):
   throw %9 : $Error
 }
 
+sil [ossa] @use_T : $@convention(thin) <T> (@in T) -> ()
+
+// CHECK-LABEL: sil [ossa] @test_checked_cast_br1 : $@convention(method) <T> (@in Any) -> () {
+// CHECK: bb0(%0 : $*Any):
+// CHECK:   [[FAILURE_DST:%.*]] = alloc_stack $Any
```

Comment:
Why allocate storage for the failure value? The value doesn't change and is not copied. The original value should simply be destroyed. During allocation, the checked_cast_br's failure value should be handled by `getReusedStorageOperand`, just like a switch_enum. Does that work?

Reply:
That works. Thanks. I updated the PR fixing this.